src/DataSourceProvider.vue

```diff
@@ -151,6 +151,26 @@ export default {
 
       this.loadDataSources(this.widgetData.appId);
     },
+    addAppToExistginRule() {
+      this.selectedDataSource.accessRules.forEach(dataSourceRule => {
+        if (dataSourceRule.appId && !dataSourceRule.appId.includes(this.widgetData.appId)) {
```

Comment:
Please rewrite the code to use early returns whenever possible as per code standards, e.g.

```
if (false) {
  return 
}

// other code
```

Reply:
@squallstar fixed